refgenconf/refgenconf.py

```diff
@@ -807,27 +834,41 @@ def chk_digest_update_child(self, genome, remote_asset_name, child_name):
         remote_asset_data = prp(remote_asset_name)
         asset = remote_asset_data["item"]
         tag = remote_asset_data["tag"]
-        asset_digest_url = construct_request_url(self.genome_server, API_ID_DIGEST).\
-            format(genome=genome, asset=asset, tag=tag)
-        try:
-            remote_digest = _download_json(asset_digest_url)
-        except DownloadJsonError:
-            return
-        try:
-            # we need to allow for missing seek_keys section so that the digest is respected even from the previously
-            # populated 'incomplete asset' from the server
-            _assert_gat_exists(self[CFG_GENOMES_KEY], genome, asset, tag,
-                               allow_incomplete=not self.is_asset_complete(genome, asset, tag))
-        except (KeyError, MissingAssetError, MissingGenomeError, MissingSeekKeyError):
-            self.update_tags(genome, asset, tag, {CFG_ASSET_CHECKSUM_KEY: remote_digest})
-            _LOGGER.info("Could not find '{}/{}:{}' digest. Populating with server data".format(genome, asset, tag))
-        else:
-            local_digest = self[CFG_GENOMES_KEY][genome][CFG_ASSETS_KEY][asset][CFG_ASSET_TAGS_KEY]\
-                [tag][CFG_ASSET_CHECKSUM_KEY]
-            if remote_digest != local_digest:
-                raise RemoteDigestMismatchError(asset, local_digest, remote_digest)
-        finally:
-            self.update_relatives_assets(genome, asset, tag, [child_name], children=True)
+
+        num_servers = 0
+        bad_servers = []
+        for server_url in self.genome_server:
```

Comment:
With multiple servers defined this method will check and update the asset relationship data multiple times (for each server).

We should not loop through all the servers here since this method is called in `refgenie` after a successful asset pull to check and update asset relationship data. Consequently, we just need to record the server that the new asset was actually pulled from (in `pull_asset` method) and pass the URL here and use just this URL.